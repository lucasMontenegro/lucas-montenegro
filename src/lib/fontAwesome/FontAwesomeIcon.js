import "lib/fontAwesome/setup"
export { FontAwesomeIcon as default } from "@fortawesome/react-fontawesome"
