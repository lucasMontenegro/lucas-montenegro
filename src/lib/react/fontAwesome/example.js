import React from "react"
import { FontAwesomeIcon } from "./index.js"
import { Route } from "react-router-dom"
function Example () {
  return (
    <div style={{ fontSize: 32 }}>
      <FontAwesomeIcon icon={[`fab`, `airbnb`]} />
      <FontAwesomeIcon icon={[`fas`, `address-card`]} />
      <FontAwesomeIcon icon={[`far`, `address-card`]} />
    </div>
  )
}
export default (<Route exact path="/react/fontAwesome" component={Example} />)