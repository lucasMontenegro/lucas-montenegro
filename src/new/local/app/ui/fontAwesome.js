const { library } = require("@fortawesome/fontawesome-svg-core")
const { fab } = require("@fortawesome/free-brands-svg-icons")
const { fas } = require("@fortawesome/free-solid-svg-icons")
const { far } = require("@fortawesome/free-regular-svg-icons")
library.add(fab, fas, far)