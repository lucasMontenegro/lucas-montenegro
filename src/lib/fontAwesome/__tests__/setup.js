describe(`lib/fontAwesome/setup`, () => {
  it(`should use the right dependency versions`, () => {
    expect(jestUtils.getDependencies([
      `@fortawesome/fontawesome-svg-core`,
      `@fortawesome/free-brands-svg-icons`,
      `@fortawesome/free-solid-svg-icons`,
      `@fortawesome/free-regular-svg-icons`,
    ])).toMatchSnapshot()
  })
  it(`should run`, () => {
    require("lib/fontAwesome/setup")
  })
})