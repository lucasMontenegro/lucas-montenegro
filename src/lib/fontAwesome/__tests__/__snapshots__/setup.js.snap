// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/fontAwesome/setup should use the right dependency versions 1`] = `
Object {
  "@fortawesome/fontawesome-svg-core": "^1.2.25",
  "@fortawesome/free-brands-svg-icons": "^5.11.2",
  "@fortawesome/free-regular-svg-icons": "^5.11.2",
  "@fortawesome/free-solid-svg-icons": "^5.11.2",
}
`;
